clippy_lints/src/no_effect.rs

```diff
@@ -39,36 +39,36 @@ fn has_no_effect(cx: &LateContext, expr: &Expr) -> bool {
         Expr_::ExprClosure(..) |
         Expr_::ExprPath(..) => true,
         Expr_::ExprIndex(ref a, ref b) |
-        Expr_::ExprBinary(_, ref a, ref b) => has_no_effect(cx, a) && has_no_effect(cx, b),
+        Expr_::ExprBinary(_, ref a, ref b) => effectless(cx, a) && effectless(cx, b),
```

Comment:
This is kind of cheating. If some name shouldn't be used because it is redundant, a synonym shouldn't be used either.